packages/graph-engine/src/nodes/series/linear.ts

```diff
@@ -0,0 +1,98 @@
+import { INodeDefinition, ToInput, ToOutput } from '../../index.js';
+import { Node } from '../../programmatic/node.js';
+import { NumberSchema } from '../../schemas/index.js';
+import { arrayOf } from '../../schemas/utils.js';
+import { setToPrecision } from '../../utils/precision.js';
+
+type LinearValue = {
+	index: number;
+	value: number;
+};
+
+export default class NodeDefinition extends Node {
+	static title = 'Linear Space';
+	static type = 'studio.tokens.series.linear';
+	static description =
+		'Creates evenly spaced numbers over a specified interval';
+
+	declare inputs: ToInput<{
+		start: number;
+		stop: number;
+		count: number;
+		precision: number;
+	}>;
+
+	declare outputs: ToOutput<{
+		array: number[];
+		indexed: LinearValue[];
+	}>;
+
+	constructor(props: INodeDefinition) {
+		super(props);
+		this.addInput('start', {
+			type: {
+				...NumberSchema,
+				default: 0
+			}
+		});
+		this.addInput('stop', {
+			type: {
+				...NumberSchema,
+				default: 1
+			}
+		});
+		this.addInput('count', {
+			type: {
+				...NumberSchema,
+				default: 5
+			}
+		});
+		this.addInput('precision', {
+			type: {
+				...NumberSchema,
+				default: 2
+			}
+		});
+		this.addOutput('array', {
+			type: arrayOf(NumberSchema)
+		});
+		this.addOutput('indexed', {
```

Comment:
I don't think we should support this indexed output to be honest, it feels like this can just be generated by another node  it does not need to be done in the node itself as its probably not used very often